### PR TITLE
Improve Message Requests UI for iOS

### PR DESCRIPTION
**Pull request by lourou**
### Changes

- Implemented a segmented controller to switch between "You might know" and "Spam" sections
- Added conditional rendering logic for different scenarios:
  - Both spam and non-spam messages present
  - Only non-spam messages present
  - Only spam messages present
  - No messages present
- Improved UI text to provide context for each scenario

### Screenshots

Attaching dark mode screenshots too:

Spamonly req button

Simulator Screenshot - iPad Air 11-inch (M2) - 2024-09-03 at 16 10 55

IMG_1287

IMG_1288

Spam Only

Closes [#178](https://github.com/ephemeraHQ/converse-app/issues/178)

### CODE REVIEW

**Comment by darickdang** [comment]
@lourou Here are some polish recommendations:

#178 - Requests bar does not match new design

**Comment by lourou** [comment]
Thanks for the feedback @darickdang I've made the changed:
- segmented controller is now smaller in height (32px instead or 40px)
- changed the bold font to semi-bold
- applied the padding as requested
- changed the information text size from 14 to 12
- used a darker gray for the zero request gray button (secondary gray)

**Comment by lourou** [comment]
iPhone, main screen:
Simulator Screenshot - iPhone 15 - 2024-09-05 at 16 40 41

iPhone, with segmented controller:
Simulator Screenshot - iPhone 15 - 2024-09-05 at 16 40 50

iPad, without segmented:
Simulator Screenshot - iPad Air 11-inch (M2) - 2024-09-05 at 16 40 55

File: screens/Navigation/ConversationRequestsListNav.ios.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      }
+    );
+  }, [account, colorScheme, allRequests]);
+
+  const navigationOptions = useCallback(
+    ({
+      navigation,
+    }: {
+      route: RouteProp<NavigationParamList, "ChatsRequests">;
+      navigation: any;
```
Comment (alexrisch) [inline screens/Navigation/ConversationRequestsListNav.ios.tsx:90]: Can we avoid anys?

File: screens/Navigation/ConversationRequestsListNav.ios.tsx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    }) => ({
+      animation: navigationAnimation as StackAnimationTypes,
+      headerTitle: clearingAll
+        ? Platform.OS === "ios"
```
Comment (alexrisch) [inline screens/Navigation/ConversationRequestsListNav.ios.tsx:94]: This is in a .ios file so no need to check these things

**Review by alexrisch — APPROVED** [review]
Approving for builds, will make some clean up tickets for comments

File: screens/Navigation/ConversationRequestsListNav.ios.tsx (hunk 2 above)
Comment (lourou) [reply screens/Navigation/ConversationRequestsListNav.ios.tsx:94]: Thanks, this is fixed!

File: screens/Navigation/ConversationRequestsListNav.ios.tsx (hunk 1 above)
Comment (lourou) [reply screens/Navigation/ConversationRequestsListNav.ios.tsx:90]: Changed to `NativeStackNavigationProp<any>;`

**Pull request merged**